### ISSUE TITLE
[$1000] Compose field's left border doesn't extend as the field expands

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17424]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open chat and start typing
2. Goto second line by pressing Shift + Return (Enter)
3. Keep Typing and goto Next line by Pressing Shift + Return (Enter)
4. Check line between + Button & type area (the left border of the compose field)

## Expected Result: 
Line Should extend along with text box size

## Actual Result: 
Line is not extending along with text box size

![image](https://user-images.githubusercontent.com/87341702/232123759-0caf2f60-ae27-4c30-9260-9d55d6d007c7.png)

## Workaround:

N/A - minimal impact

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
<img width="960" alt="Untitled" src="https://user-images.githubusercontent.com/43996225/231903879-b4023b89-8294-4e59-98eb-e35353fc19f1.png">
![Screenshot 2023-04-13 at 2 41 58 PM](https://user-images.githubusercontent.com/43996225/231903887-ec76600d-f52e-4318-9e0e-2c0f9540e0de.png)

https://user-images.githubusercontent.com/43996225/231903889-b83833eb-198e-4ffc-9bac-79b10c813894.mov

**Expensify/Expensify Issue URL:**
**Issue reported by:** @corange-lab
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1681377556974939

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019ff38561f983d47e</li>
        <li>Upwork Job ID: 1646855304295243776</li>
        <li>Last Price Increase: 2023-04-14</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @conorpendergrast (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by conorpendergrast** [comment]
@kavimuru this issue title seems... wrong 😆

**Comment by corange-lab** [comment]
> @kavimuru this issue title seems... wrong 😆

@conorpendergrast i wrote that title and i don't have word for it so ..

**Comment by conorpendergrast** [comment]
Oh hah, I just spotted that it's in the Slack thread about this 😄

**Comment by conorpendergrast** [comment]
I'm going to re-title this to make it a little clearer, but I did enjoy that previous title. Reproduced

![image](https://user-images.githubusercontent.com/5802477/232045239-bd73a627-7d15-4046-89f0-ecee0c0f71c4.png)

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~019ff38561f983d47e

**Comment by MelvinBot** [comment]
Current assignee @conorpendergrast is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When composer has multiline text at that time right border beside Plus icon does not expand vertically.

### What is the root cause of that problem?

This is the parent view that consist all three button i.e. collapse, expand, and addAction
https://github.com/Expensify/App/blob/ab9b7d9ef1fb1656f5ccfc28ca4a68820fa373b4/src/pages/home/report/ReportActionCompose/index.js#L764-L768

Plus button shows right border via View having `styles.chatItemAttachBorder`  as shown below:
https://github.com/Expensify/App/blob/ab9b7d9ef1fb1656f5ccfc28ca4a68820fa373b4/src/pages/home/report/ReportActionCompose/index.js#L807

`chatItemAttachBorder` consist below style:
https://github.com/Expensify/App/blob/ab9b7d9ef1fb1656f5ccfc28ca4a68820fa373b4/src/styles/styles.js#L1587-L1590

So during multiline plus button view does not expand vertically so line does not show. This is the root cause of the problem.

### What changes do you think we should make in order to solve the problem?
We have to do below changes to solve this
1. Remove View from line 807 as shown below
```javascript
<View style={styles.chatItemAttachBorder}>    // *** Remove this
   .... // Keep this content
</View>.     // *** Remove this
```

3. Apply `styles.chatItemAttachBorder` to its parent View that starts at line 764 as shown below:
```javascript
 <View style={[ 
     styles.dFlex, styles.flexColumn, 
     (this.state.isFullComposerAvailable || this.props.isComposerFullSize) ? styles.justifyContentBetween : styles.justifyContentEnd, 
    styles.chatItemAttachBorder,   // *** Add this
 ]} 
 > 
```

**Note:** We can rename `styles.chatItemAttachBorder` if wish, otherwise it is ok. It consist style for right border.

So this will solve the issue. It is working as expected in all platform. Sample video added in Results section below.

### What alternative solutions did you explore? (Optional)
None

<details>
<summary>Results</summary>

https://user-images.githubusercontent.com/7823358/232017357-d7f7142a-feac-4587-ba19-5f4fc8d0333c.mov

</details>

**Comment by AmjedNazzal** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Line in the composer doesn't extend as the user presses shift enter

### What is the root cause of that problem?

This line is created by using a borderRight. this border is added with "styles.chatItemAttachBorder" but originally only added to the View element of the actionButton which does not wrap the entire element that will expand with the textArea 

### What changes do you think we should make in order to solve the problem?

We can move the style to a higher level, that way the border is attached to the entire element and will expand properly

<AttachmentPicker>
    {({openPicker}) => (
        <>
            <View style={[
                styles.chatItemAttachBorder,
                styles.dFlex, styles.flexColumn,
                (this.state.isFullComposerAvailable || this.props.isComposerFullSize) ? styles.justifyContentBetween : styles.justifyContentEnd,
            ]}
            >
           
            

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by AmjedNazzal** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Line in the composer doesn't extend as the user presses shift enter

### What is the root cause of that problem?

This line is created by using a borderRight. this border is added with "styles.chatItemAttachBorder" but originally only added to the View element of the actionButton which does not wrap the entire element that will expand with the textArea 

### What changes do you think we should make in order to solve the problem?

We can move the style to a higher level, that way the border is attached to the entire element and will expand properly

<AttachmentPicker>
    {({openPicker}) => (
        <>
            <View style={[
                styles.chatItemAttachBorder,
                styles.dFlex, styles.flexColumn,
                (this.state.isFullComposerAvailable || this.props.isComposerFullSize) ? styles.justifyContentBetween : styles.justifyContentEnd,
            ]}
            >

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by AmjedNazzal** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Line in the composer doesn't extend as the user presses shift enter

### What is the root cause of that problem?

This line is created by using a borderRight. this border is added with "styles.chatItemAttachBorder" but originally only added to the View element of the actionButton which does not wrap the entire element that will expand with the textArea 

### What changes do you think we should make in order to solve the problem?

We can move the style to a higher level, that way the border is attached to the entire element and will expand properly

```
<AttachmentPicker>
    {({openPicker}) => (
        <>
            <View style={[
                styles.chatItemAttachBorder,
                styles.dFlex, styles.flexColumn,
                (this.state.isFullComposerAvailable || this.props.isComposerFullSize) ? styles.justifyContentBetween : styles.justifyContentEnd,
            ]}
            >
```

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by mollfpr** [comment]
Thank you guys for the proposals!

While everyone has the same correct solution and @PrashantMangukiya [proposal](https://github.com/Expensify/App/issues/17424#issuecomment-1508443896) is the first to post it right after the `Help Wanted` label. @aldo-expensify, we can go with @PrashantMangukiya's proposal!

<details>
<summary>Result</summary>
<img width="746" alt="Screenshot 2023-04-15 at 01 13 24" src="https://user-images.githubusercontent.com/25520267/232124591-2da8765a-0f8f-4c39-beaa-3e2705c58fad.png">
<img width="737" alt="Screenshot 2023-04-15 at 01 13 17" src="https://user-images.githubusercontent.com/25520267/232124595-82cf59ca-e095-4d2f-88f9-2bd21f8c760b.png">

</details>

🎀 👀 🎀 C+ reviewed!

**Comment by MelvinBot** [comment]
📣 @PrashantMangukiya You have been assigned to this job by @aldo-expensify!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by PrashantMangukiya** [comment]
@mollfpr  @aldo-expensify Thank you. Accepted job on Upwork. I will prepare and submit PR within two hours asap.

**Comment by PrashantMangukiya** [comment]
@mollfpr @aldo-expensify PR is ready for review. Thank you.

**Comment by corange-lab** [comment]
@mollfpr @aldo-expensify can i also apply on upwork?

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.1-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17464


If no regressions arise, payment will be issued on 2023-04-26. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mollfpr] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@conorpendergrast] Determine if we should create a regression test for this bug.
- [ ] [@mollfpr] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@conorpendergrast] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by PrashantMangukiya** [comment]
Ping for Upwork.

**Comment by PrashantMangukiya** [comment]
@conorpendergrast Ping for Upwork.

**Comment by corange-lab** [comment]
@conorpendergrast i am waiting for response on upwork.

**Comment by aldo-expensify** [comment]
Conor is out of office and he will be back early next week to process your payments

**Comment by corange-lab** [comment]
@aldo-expensify Thanks for giving update.

**Comment by conorpendergrast** [comment]
Thanks all, and appreciate you applying via Upwork - makes it a lot easier! 

Assigned: April 14th 8:18pm
PR merged: April 17th 8:15pm
That's less than 3 business days, and so C and C+ are eligible for 50% bonus

**Comment by PrashantMangukiya** [comment]
@conorpendergrast Received Thank you.

**Comment by corange-lab** [comment]
> * Invited and paid @PrashantMangukiya.
> * Invited @mollfpr to the job in Upwork.
> * @corange-lab I'm really sorry, I wasn't able to find your application on Upwork, and I haven't been able to find your Upwork profile! Can you either [apply here](https://www.upwork.com/ab/applicants/1646855304295243776/job-details), or reply with your Upwork profile URL and I'll invite you? Thanks!

@conorpendergrast 
https://www.upwork.com/freelancers/~0182d5b08109999ffb
Thanks for response

**Comment by corange-lab** [comment]
@conorpendergrast Accepted Thanks.

**Comment by aldo-expensify** [comment]
Friendly ping @mollfpr on completing: https://github.com/Expensify/App/issues/17424#issuecomment-1515078000

**Comment by mollfpr** [comment]
@aldo-expensify Sorry I miss this; I'll complete the checklist today 🙏

**Comment by mollfpr** [comment]
> [@mollfpr] The PR that introduced the bug has been identified. Link to the PR:
> [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

This issue has existed since the first implementation of the attachment button inside the composer https://github.com/Expensify/App/pull/399. Then it got refactored here https://github.com/Expensify/App/pull/13645, where it's not part of the button anymore.

So I don't think any of that PR is causing the bug, but now we see it as unsuitable.

> [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

The checklist is good for now!

**Comment by mollfpr** [comment]
>[@mollfpr] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

For the regression test, I'll propose to have it under [Compose box - Text Input](https://expensify.testrail.io/index.php?/cases/view/1971009). At _**Enter 3+ lines of text, and select/highlight the text**_, we can add _**Verify that the border after the action button is extended accordingly**_.

👍 or 👎?

**Comment by aldo-expensify** [comment]
That sounds perfect for me ^

**Comment by conorpendergrast** [comment]
Sounds good to me too - I'll create that regression test
